cmd/up/exec.go

```diff
@@ -16,27 +16,268 @@ package up
 import (
 	"bytes"
 	"context"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
+	"io"
 	"os"
+	"os/exec"
+	"path/filepath"
 	"strings"
 
+	"github.com/creack/pty"
 	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
 	"github.com/okteto/okteto/pkg/config"
+	"github.com/okteto/okteto/pkg/constants"
 	oktetoErrors "github.com/okteto/okteto/pkg/errors"
 	"github.com/okteto/okteto/pkg/k8s/apps"
-	"github.com/okteto/okteto/pkg/k8s/exec"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	k8sExec "github.com/okteto/okteto/pkg/k8s/exec"
 	"github.com/okteto/okteto/pkg/k8s/pods"
 	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/registry"
 	"github.com/okteto/okteto/pkg/ssh"
+	"github.com/okteto/okteto/pkg/types"
+	"k8s.io/client-go/kubernetes"
 )
 
+type devExecutor interface {
+	runCommand(ctx context.Context, cmd []string) error
+}
+
+type hybridExecutor struct {
+	workdir string
+	envs    []string
+}
+
+func (he *hybridExecutor) runCommand(_ context.Context, cmd []string) error {
+	c := exec.Command("bash", "-c", strings.Join(cmd, " "))
```

Comment:
I think the bash here could break other scenarios (Windows, docker images with alpine...) Should we just run it without the `bash -c`?

Reply:
I followed what we have now where bash is always used unless remote deploys. https://github.com/okteto/okteto/blob/master/cmd/utils/executor/executor.go#L61. It is OK?

Reply:
On the deploy Windows users can omit the bash/sh by setting the flag `--no-bash`. Otherwise we were redirecting them to WSL. Here is where we do that:  https://github.com/okteto/okteto/blob/master/cmd/utils/executor/executor.go#L78

Reply:
done fbfde67a4651acf3c7676ff3f1b71be1788d94d1